framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -541,7 +517,14 @@ public Boolean visitDeclared_Typevar(AnnotatedDeclaredType subtype, AnnotatedTyp
 
     @Override
     public Boolean visitDeclared_Union(AnnotatedDeclaredType subtype, AnnotatedUnionType supertype, VisitHistory visited) {
-        return visitUnionSupertype(subtype, supertype, visited);
+        Types types = checker.getTypeUtils();
+        for (AnnotatedDeclaredType supertypeAltern : supertype.getAlternatives()) {
+            if (TypesUtils.isErasedSubtype(types, subtype.getUnderlyingType(), supertypeAltern.getUnderlyingType())
```

Comment:
I agree on I_I and noted that in my first comment.
Primitives need the boxing/unboxing handling and that would cover the Primitive cases.
Logically, subtyping on annotated types should be stronger than subtyping on the underlying type. Putting this logic together into isSubtype would make this clear and prevent forgetting about special cases in different invocations.
If we keep the two test separate, in a new location someone might forget about boxed/unboxed types needing special treatment.
If we put all this logic into isSubtype, there is no repetition and the purpose should be clearer.